### PR TITLE
Remove mono-specific runtime code

### PR DESCRIPTION
**Pull request by shaan1337**

### CODE REVIEW

**Comment by shaan1337** [comment]
@jageall @pgermishuys I've obtained the following results with following project: [checkpointtest.zip](https://github.com/EventStore/EventStore/files/3977528/checkpointtest.zip) on ubuntu 16.04:

``` ini

BenchmarkDotNet=v0.12.0, OS=ubuntu 16.04
Intel Core i7-7700HQ CPU 2.80GHz (Kaby Lake), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=3.1.100
  [Host]     : .NET Core 3.1.0 (CoreCLR 4.700.19.56106, CoreFX 4.700.19.56202), X64 RyuJIT
  DefaultJob : .NET Core 3.1.0 (CoreCLR 4.700.19.56106, CoreFX 4.700.19.56202), X64 RyuJIT


```
|                 Method |     Mean |     Error |    StdDev |
|----------------------- |---------:|----------:|----------:|
| MemoryMappedCheckpoint | 8.919 ms | 0.1755 ms | 0.1724 ms |
|         FileCheckpoint | 8.125 ms | 0.1540 ms | 0.1440 ms |

File: src/EventStore.Transport.Http/Server/HttpAsyncServer.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public sealed class HttpAsyncServer {
 		}
 
 		private void TryAddAcl(string address) {
-			if (Runtime.IsMono)
+			if (!Runtime.IsWindows)
```
[the remark this answers is not in the document]
Comment (shaan1337) [reply src/EventStore.Transport.Http/Server/HttpAsyncServer.cs]: @pgermishuys ok, i've removed it in this new commit: fc05ae2c4c44da663376e2b88ddb8f30e3c85bef

**Comment by arwinneil** [comment]
@shaan1337 Benchmark results on **Windows 10**

````
BenchmarkDotNet=v0.12.0, OS=Windows 10.0.18362
Intel Core i5-7200U CPU 2.50GHz (Kaby Lake), 1 CPU, 4 logical and 2 physical cores
.NET Core SDK=3.1.100
  [Host]     : .NET Core 3.1.0 (CoreCLR 4.700.19.56402, CoreFX 4.700.19.56404), X64 RyuJIT
  DefaultJob : .NET Core 3.1.0 (CoreCLR 4.700.19.56402, CoreFX 4.700.19.56404), X64 RyuJIT
````

|                 Method |     Mean |     Error |    StdDev |
|----------------------- |---------:|----------:|----------:|
| MemoryMappedCheckpoint | 1.467 ms | 0.0579 ms | 0.1632 ms |
|         FileCheckpoint | 2.980 ms | 0.0959 ms | 0.2782 ms |

File: src/EventStore.Core/VNodeBuilder.cs [hunk 2]
```diff
@@ -1492,20 +1492,12 @@ public abstract class VNodeBuilder {
...
-					truncateChk = new MemoryMappedFileCheckpoint(truncateCheckFilename, Checkpoint.Truncate,
-						cached: true, initValue: -1);
-				} else {
-					writerChk = new FileCheckpoint(writerCheckFilename, Checkpoint.Writer, cached: true);
-					chaserChk = new FileCheckpoint(chaserCheckFilename, Checkpoint.Chaser, cached: true);
-					epochChk = new FileCheckpoint(epochCheckFilename, Checkpoint.Epoch, cached: true, initValue: -1);
-					truncateChk = new FileCheckpoint(truncateCheckFilename, Checkpoint.Truncate, cached: true,
-						initValue: -1);
-				}
+				writerChk = new MemoryMappedFileCheckpoint(writerCheckFilename, Checkpoint.Writer, cached: true);
```
[the remark this answers is not in the document]
Comment (shaan1337) [reply src/EventStore.Core/VNodeBuilder.cs]: @pgermishuys @jageall benchmarks added. Thanks to @arwinneil for running the benchmarks on Windows.

**Comment by arwinneil** [comment]
Benchmark Results on Amazon Linux 2
````
BenchmarkDotNet=v0.12.0, OS=amzn 2
Intel Xeon CPU E5-2676 v3 2.40GHz, 1 CPU, 1 logical core and 1 physical core
.NET Core SDK=3.1.100
  [Host]     : .NET Core 3.1.0 (CoreCLR 4.700.19.56402, CoreFX 4.700.19.56404), X64 RyuJIT
  DefaultJob : .NET Core 3.1.0 (CoreCLR 4.700.19.56402, CoreFX 4.700.19.56404), X64 RyuJIT
````
|                 Method |     Mean |    Error |   StdDev |
|----------------------- |---------:|---------:|---------:|
| MemoryMappedCheckpoint | 562.4 us | 11.08 us | 23.85 us |
|         FileCheckpoint | 573.7 us | 11.38 us | 29.57 us |

**Pull request merged**